### PR TITLE
Spinner 2

### LINKED ISSUE #2529: Spinner v2
Details to be filled by @dplsek 
```[tasklist]
### Tasks
- [ ] Density (HD, MD, LD, TD)
- [ ] Mode (Light, Dark)
```

```[tasklist]
### Deliverables
- [ ] https://github.com/jpmorganchase/salt-ds/issues/2590
- [ ] https://github.com/jpmorganchase/salt-ds/issues/2591
```

```[tasklist]
### Features
```

### PR DESCRIPTION
**Pull request by lilyvc**
closes #2529 

This PR 
- adds size `small` to Spinner and updates the bar thickness across sizes
- adds `--salt-size-bar-small` to theme

### CODE REVIEW

File: packages/core/stories/spinner/spinner.doc.stories.mdx [hunk 1]
```diff
@@ -24,20 +24,20 @@ Improve accessibility by customizing the `aria-label` to provide additional cont
...
+## Nested
 
-Set the `size` prop to `"large"` to use the large Spinner.
+Set the `size` prop to `"nested"` to use the nested Spinner.
 
 Large Spinners can be beneficial for full screen loading experiences; while default Spinners can be reserved for widget loading experiences.
 
 <Canvas>
-  <Story id="core-spinner--large" />
+  <Story id="core-spinner--nested" />
```
Comment (DavieReid) [inline packages/core/stories/spinner/spinner.doc.stories.mdx:40]: Say it with me "99% of the time, if the example has to change to accommodate a code change then it's a breaking change"

P.S. 67% of statistics are mad up on the spot 😉

File: packages/core/stories/spinner/spinner.stories.tsx [hunk 2]
```diff
@@ -78,7 +87,7 @@ export const Loading: StoryFn = () => {
         <Spinner
           style={{ margin: "auto" }}
           aria-label="Panel is loading"
-          size="large"
+          size="small"
```
Comment (lilyvc) [inline packages/core/stories/spinner/spinner.stories.tsx:90]:
```suggestion
          size="large"
```

**Review by feliciackh — DISMISSED** [review]
LGTM, though the small spinner almost look out of place given the sizes of default & large

**Comment by lilyvc** [comment]
> LGTM, though the small spinner almost look out of place given the sizes of default & large

thanks for the review. I kind of know what you mean but I guess there is a use case

**Review by JoanaMMoreira — APPROVED** [review]
Do we also want to add a site example for size small?

**Comment by lilyvc** [comment]
> Do we also want to add a site example for size small?

thanks for the review :)  yes - will do in a follow up PR. The text for the example is still being worked on

**Pull request merged**